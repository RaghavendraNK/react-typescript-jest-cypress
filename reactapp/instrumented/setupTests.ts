function cov_1nrbe6xq6y() {
    var path =
        '/home/raghavendra/Documents/Github/react-typescript-jest-cypress/src/setupTests.ts';
    var hash = '61624976bc4c0f3268a508313704a486eff5dc1d';
    var global = new Function('return this')();
    var gcv = '__coverage__';
    var coverageData = {
        path:
            '/home/raghavendra/Documents/Github/react-typescript-jest-cypress/src/setupTests.ts',
        statementMap: {},
        fnMap: {},
        branchMap: {},
        s: {},
        f: {},
        b: {},
        _coverageSchema: '1a1c01bbd47fc00a2c39e90264f33305004495a9',
        hash: '61624976bc4c0f3268a508313704a486eff5dc1d',
    };
    var coverage = global[gcv] || (global[gcv] = {});

    if (!coverage[path] || coverage[path].hash !== hash) {
        coverage[path] = coverageData;
    }

    var actualCoverage = coverage[path];
    {
        // @ts-ignore
        cov_1nrbe6xq6y = function () {
            return actualCoverage;
        };
    }
    return actualCoverage;
}

cov_1nrbe6xq6y();
// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import '@testing-library/jest-dom';
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbInNldHVwVGVzdHMudHMiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6Ijs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7O0FBZVk7Ozs7Ozs7OztBQWZaO0FBQ0E7QUFDQTtBQUNBO0FBQ0EsT0FBTywyQkFBUCIsInNvdXJjZXNDb250ZW50IjpbIi8vIGplc3QtZG9tIGFkZHMgY3VzdG9tIGplc3QgbWF0Y2hlcnMgZm9yIGFzc2VydGluZyBvbiBET00gbm9kZXMuXG4vLyBhbGxvd3MgeW91IHRvIGRvIHRoaW5ncyBsaWtlOlxuLy8gZXhwZWN0KGVsZW1lbnQpLnRvSGF2ZVRleHRDb250ZW50KC9yZWFjdC9pKVxuLy8gbGVhcm4gbW9yZTogaHR0cHM6Ly9naXRodWIuY29tL3Rlc3RpbmctbGlicmFyeS9qZXN0LWRvbVxuaW1wb3J0ICdAdGVzdGluZy1saWJyYXJ5L2plc3QtZG9tJztcbiJdfQ==
