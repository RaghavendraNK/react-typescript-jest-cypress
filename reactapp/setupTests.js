import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-17-updated';
import sinon from 'sinon';
import { expect } from 'chai';
export const mockHistoryPush = jest.fn();

configure({ adapter: new Adapter() });
global.expect = expect;
global.sinon = sinon;

jest.mock('react-i18next', () => ({
    useTranslation: () => ({ t: (key) => key }),
}));

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useHistory: () => ({
        push: mockHistoryPush,
    }),
}));
