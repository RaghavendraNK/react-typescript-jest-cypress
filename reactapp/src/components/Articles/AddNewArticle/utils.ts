import { FormListType } from 'slick-react-ui-components';

const formList: FormListType[] = [
    {
        sizes: { xs: 12, sm: 12, md: 12, lg: 12 },
        cols: [
            {
                id: 'title',
                colcss: 'page-form-col',
                name: 'title',
                type: 'TextField',
                variant: 'outlined',
                label: 'Title',
            },
        ],
    },
    {
        sizes: { xs: 12, sm: 12, md: 12, lg: 12 },
        cols: [
            {
                id: 'description',
                colcss: 'page-form-col',
                name: 'description',
                type: 'TextField',
                variant: 'outlined',
                label: 'Description',
                multiline: true,
                rows: 5,
            },
        ],
    },
    {
        sizes: { xs: 12, sm: 12, md: 12, lg: 12 },
        cols: [
            {
                colcss: 'page-form-col',
                name: 'upload',
                type: 'Upload',
            },
        ],
    },
];

const ActionTypes = {
    CLEAR_FORM: 'CLEAR_FORM',
};

const initialValues = {
    title: '',
    description: '',
    imgBase64: '',
};
export { formList, ActionTypes, initialValues };
