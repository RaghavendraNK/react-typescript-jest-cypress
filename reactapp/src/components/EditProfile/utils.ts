import { FormListType } from 'slick-react-ui-components';

const formList: FormListType[] = [
    {
        sizes: { xs: 12, sm: 12, md: 12, lg: 12 },
        cols: [
            {
                id: 'name',
                colcss: 'page-form-col',
                name: 'name',
                type: 'TextField',
                variant: 'outlined',
                label: 'Name',
            },
        ],
    },
    {
        sizes: { xs: 12, sm: 12, md: 6, lg: 6 },
        cols: [
            {
                id: 'oldPassword',
                colcss: 'page-form-col',
                name: 'oldPassword',
                variant: 'outlined',
                label: 'Old Password',
                type: 'password',
            },
            {
                id: 'newPassword',
                colcss: 'page-form-col',
                name: 'newPassword',
                variant: 'outlined',
                label: 'New Password',
                type: 'password',
            },
        ],
    },
    {
        sizes: { xs: 12, sm: 12, md: 12, lg: 12 },
        cols: [
            {
                colcss: 'page-form-col',
                name: 'upload',
                type: 'Upload',
            },
        ],
    },
];

const ActionTypes = {
    CLEAR_FORM: 'CLEAR_FORM',
};

const initialValues = {
    name: '',
    imgBase64: '',
};
export { formList, ActionTypes, initialValues };
