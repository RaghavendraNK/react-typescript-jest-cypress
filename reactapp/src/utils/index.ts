export const language = ['en'];
export const setlanguage = 'en';
