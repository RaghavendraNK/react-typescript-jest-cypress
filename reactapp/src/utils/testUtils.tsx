import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

export const MockProvider = (props: { children: JSX.Element }): JSX.Element => {
    const mockstore = configureStore();
    let store = mockstore();
    return <Provider store={store}>{props.children}</Provider>;
};
