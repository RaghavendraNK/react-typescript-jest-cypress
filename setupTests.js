import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import sinon from 'sinon';
import { expect } from 'chai';

configure({ adapter: new Adapter() });
global.expect = expect;
global.sinon = sinon;
