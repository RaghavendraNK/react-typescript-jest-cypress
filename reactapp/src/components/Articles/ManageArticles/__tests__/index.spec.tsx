import React from 'react';
import { mount, ReactWrapper } from 'enzyme';
import { BreadCrumbs, GridContainer } from 'slick-react-ui-components';

import { ManageArticles, paths } from '../index';
import { mockHistoryPush } from '../../../../../setupTests';
import { MockProvider } from '../../../../utils/testUtils';

describe('ManageArticles', () => {
    let wrapper: ReactWrapper;
    beforeAll(() => {
        wrapper = mount(
            <MockProvider>
                <ManageArticles />
            </MockProvider>
        );
    });
    it('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
    describe('DOM', () => {
        it('should render BreadCrumbs', () => {
            const breadCrumbs = wrapper.find(BreadCrumbs);
            expect(breadCrumbs).toHaveLength(1);
            expect(breadCrumbs.prop('paths')).toEqual(paths);
            breadCrumbs.props().handleClick('/path');
            expect(mockHistoryPush).toHaveBeenCalledWith('/path');
        });

        it('should render GridContainer', () => {
            const component = wrapper.find(GridContainer);
            expect(component).toHaveLength(1);
        });
    });
});
