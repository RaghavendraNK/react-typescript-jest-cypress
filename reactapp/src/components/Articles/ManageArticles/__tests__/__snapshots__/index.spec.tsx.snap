// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ManageArticles should match snapshot 1`] = `
<MockProvider>
  <Provider
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <ManageArticles>
      <div
        id="manageArticles"
      >
        <Component
          handleClick={[Function]}
          paths={
            Array [
              Object {
                "id": "homeLink",
                "label": "Home",
                "path": "/home",
              },
              Object {
                "id": "manageArticles",
                "label": "Manage articles",
                "path": "/manageArticles",
              },
            ]
          }
        >
          <div
            id="RRUCBreadCrumbs"
          >
            <div
              className="breadcrumbs"
            >
              <f
                id="homeLink"
                label="Home"
                onClick={[Function]}
                variant="link"
              >
                <div
                  id="RRUCButton"
                >
                  <span
                    className="btn-link"
                    onClick={[Function]}
                  >
                    Home
                  </span>
                </div>
              </f>
              <span
                className="bradcrumbs-divider"
              >
                /
              </span>
              <div
                className="greyLink"
              >
                Manage articles
              </div>
            </div>
          </div>
        </Component>
        <ForwardRef
          description="en:managearticle-description"
          header="en:managearticle-header"
        >
          <Component
            description="en:managearticle-description"
            forwardedRef={null}
            header="en:managearticle-header"
          >
            <InnerLoadable
              description="en:managearticle-description"
              forwardedRef={null}
              header="en:managearticle-header"
            />
          </Component>
        </ForwardRef>
        <div
          className="cards-section"
        >
          <k
            containerCss=""
            itemCss=""
            name="manageArticles"
            sizes={
              Object {
                "lg": 4,
                "md": 4,
                "sm": 12,
                "xs": 12,
              }
            }
            spacing={2}
          >
            <WithStyles(ForwardRef(Grid))
              className=""
              container={true}
              id="RRUCGridContainer"
              spacing={2}
            >
              <ForwardRef(Grid)
                className=""
                classes={
                  Object {
                    "align-content-xs-center": "MuiGrid-align-content-xs-center",
                    "align-content-xs-flex-end": "MuiGrid-align-content-xs-flex-end",
                    "align-content-xs-flex-start": "MuiGrid-align-content-xs-flex-start",
                    "align-content-xs-space-around": "MuiGrid-align-content-xs-space-around",
                    "align-content-xs-space-between": "MuiGrid-align-content-xs-space-between",
                    "align-items-xs-baseline": "MuiGrid-align-items-xs-baseline",
                    "align-items-xs-center": "MuiGrid-align-items-xs-center",
                    "align-items-xs-flex-end": "MuiGrid-align-items-xs-flex-end",
                    "align-items-xs-flex-start": "MuiGrid-align-items-xs-flex-start",
                    "container": "MuiGrid-container",
                    "direction-xs-column": "MuiGrid-direction-xs-column",
                    "direction-xs-column-reverse": "MuiGrid-direction-xs-column-reverse",
                    "direction-xs-row-reverse": "MuiGrid-direction-xs-row-reverse",
                    "grid-lg-1": "MuiGrid-grid-lg-1",
                    "grid-lg-10": "MuiGrid-grid-lg-10",
                    "grid-lg-11": "MuiGrid-grid-lg-11",
                    "grid-lg-12": "MuiGrid-grid-lg-12",
                    "grid-lg-2": "MuiGrid-grid-lg-2",
                    "grid-lg-3": "MuiGrid-grid-lg-3",
                    "grid-lg-4": "MuiGrid-grid-lg-4",
                    "grid-lg-5": "MuiGrid-grid-lg-5",
                    "grid-lg-6": "MuiGrid-grid-lg-6",
                    "grid-lg-7": "MuiGrid-grid-lg-7",
                    "grid-lg-8": "MuiGrid-grid-lg-8",
                    "grid-lg-9": "MuiGrid-grid-lg-9",
                    "grid-lg-auto": "MuiGrid-grid-lg-auto",
                    "grid-lg-true": "MuiGrid-grid-lg-true",
                    "grid-md-1": "MuiGrid-grid-md-1",
                    "grid-md-10": "MuiGrid-grid-md-10",
                    "grid-md-11": "MuiGrid-grid-md-11",
                    "grid-md-12": "MuiGrid-grid-md-12",
                    "grid-md-2": "MuiGrid-grid-md-2",
                    "grid-md-3": "MuiGrid-grid-md-3",
                    "grid-md-4": "MuiGrid-grid-md-4",
                    "grid-md-5": "MuiGrid-grid-md-5",
                    "grid-md-6": "MuiGrid-grid-md-6",
                    "grid-md-7": "MuiGrid-grid-md-7",
                    "grid-md-8": "MuiGrid-grid-md-8",
                    "grid-md-9": "MuiGrid-grid-md-9",
                    "grid-md-auto": "MuiGrid-grid-md-auto",
                    "grid-md-true": "MuiGrid-grid-md-true",
                    "grid-sm-1": "MuiGrid-grid-sm-1",
                    "grid-sm-10": "MuiGrid-grid-sm-10",
                    "grid-sm-11": "MuiGrid-grid-sm-11",
                    "grid-sm-12": "MuiGrid-grid-sm-12",
                    "grid-sm-2": "MuiGrid-grid-sm-2",
                    "grid-sm-3": "MuiGrid-grid-sm-3",
                    "grid-sm-4": "MuiGrid-grid-sm-4",
                    "grid-sm-5": "MuiGrid-grid-sm-5",
                    "grid-sm-6": "MuiGrid-grid-sm-6",
                    "grid-sm-7": "MuiGrid-grid-sm-7",
                    "grid-sm-8": "MuiGrid-grid-sm-8",
                    "grid-sm-9": "MuiGrid-grid-sm-9",
                    "grid-sm-auto": "MuiGrid-grid-sm-auto",
                    "grid-sm-true": "MuiGrid-grid-sm-true",
                    "grid-xl-1": "MuiGrid-grid-xl-1",
                    "grid-xl-10": "MuiGrid-grid-xl-10",
                    "grid-xl-11": "MuiGrid-grid-xl-11",
                    "grid-xl-12": "MuiGrid-grid-xl-12",
                    "grid-xl-2": "MuiGrid-grid-xl-2",
                    "grid-xl-3": "MuiGrid-grid-xl-3",
                    "grid-xl-4": "MuiGrid-grid-xl-4",
                    "grid-xl-5": "MuiGrid-grid-xl-5",
                    "grid-xl-6": "MuiGrid-grid-xl-6",
                    "grid-xl-7": "MuiGrid-grid-xl-7",
                    "grid-xl-8": "MuiGrid-grid-xl-8",
                    "grid-xl-9": "MuiGrid-grid-xl-9",
                    "grid-xl-auto": "MuiGrid-grid-xl-auto",
                    "grid-xl-true": "MuiGrid-grid-xl-true",
                    "grid-xs-1": "MuiGrid-grid-xs-1",
                    "grid-xs-10": "MuiGrid-grid-xs-10",
                    "grid-xs-11": "MuiGrid-grid-xs-11",
                    "grid-xs-12": "MuiGrid-grid-xs-12",
                    "grid-xs-2": "MuiGrid-grid-xs-2",
                    "grid-xs-3": "MuiGrid-grid-xs-3",
                    "grid-xs-4": "MuiGrid-grid-xs-4",
                    "grid-xs-5": "MuiGrid-grid-xs-5",
                    "grid-xs-6": "MuiGrid-grid-xs-6",
                    "grid-xs-7": "MuiGrid-grid-xs-7",
                    "grid-xs-8": "MuiGrid-grid-xs-8",
                    "grid-xs-9": "MuiGrid-grid-xs-9",
                    "grid-xs-auto": "MuiGrid-grid-xs-auto",
                    "grid-xs-true": "MuiGrid-grid-xs-true",
                    "item": "MuiGrid-item",
                    "justify-xs-center": "MuiGrid-justify-xs-center",
                    "justify-xs-flex-end": "MuiGrid-justify-xs-flex-end",
                    "justify-xs-space-around": "MuiGrid-justify-xs-space-around",
                    "justify-xs-space-between": "MuiGrid-justify-xs-space-between",
                    "justify-xs-space-evenly": "MuiGrid-justify-xs-space-evenly",
                    "root": "MuiGrid-root",
                    "spacing-xs-1": "MuiGrid-spacing-xs-1",
                    "spacing-xs-10": "MuiGrid-spacing-xs-10",
                    "spacing-xs-2": "MuiGrid-spacing-xs-2",
                    "spacing-xs-3": "MuiGrid-spacing-xs-3",
                    "spacing-xs-4": "MuiGrid-spacing-xs-4",
                    "spacing-xs-5": "MuiGrid-spacing-xs-5",
                    "spacing-xs-6": "MuiGrid-spacing-xs-6",
                    "spacing-xs-7": "MuiGrid-spacing-xs-7",
                    "spacing-xs-8": "MuiGrid-spacing-xs-8",
                    "spacing-xs-9": "MuiGrid-spacing-xs-9",
                    "wrap-xs-nowrap": "MuiGrid-wrap-xs-nowrap",
                    "wrap-xs-wrap-reverse": "MuiGrid-wrap-xs-wrap-reverse",
                    "zeroMinWidth": "MuiGrid-zeroMinWidth",
                  }
                }
                container={true}
                id="RRUCGridContainer"
                spacing={2}
              >
                <div
                  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
                  id="RRUCGridContainer"
                />
              </ForwardRef(Grid)>
            </WithStyles(ForwardRef(Grid))>
          </k>
        </div>
      </div>
    </ManageArticles>
  </Provider>
</MockProvider>
`;
